### DISCUSSION TITLE
how to use with windows.

### DISCUSSION THREAD

**Discussion by Trcer** [spatie/image-optimizer#160] [Q&A]
hey, it says to download various tools like jpegotpem etc... to optimize images. how do i install it on windows?

**Comment by nmeri17** [comment]
@Trcer how did you resolve this?

**Reply by Tyjow** [reply]
Hi guys,

To install **jpegoptim** on Windows for example, simply download here the bin : https://sourceforge.net/projects/jpegoptim/
Then unzip it in a desired folder in your system for example : D:\mySuperBinFolder
Then you can use it in your project like that : 

```
$optimizer = new OptimizerChain();
$optimizer->setTimeout(10);
$optimizer->addOptimizer((new Jpegoptim([
    '--all-progressive',
]))->setBinaryPath('D:\mySuperBinFolder'));
```
